### PR TITLE
Add HLRC support for delete policy api

### PR DESCRIPTION
**Pull request by martijnvg**
This PR also adds HLRC docs.

Relates to #32789

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/EnrichDocumentationIT.java [hunk 1]
```diff
@@ -77,4 +78,57 @@ public void onFailure(Exception e) {
...
+        boolean isAcknowledged =
+            deletePolicyResponse.isAcknowledged(); // <1>
+        // end::enrich-delete-policy-response
+
+        // tag::enrich-delete-policy-execute-listener
+        ActionListener<AcknowledgedResponse> listener = new ActionListener<>() {
+            @Override
+            public void onResponse(AcknowledgedResponse response) { // <1>
+                boolean isAcknowledged =
+                    deletePolicyResponse.isAcknowledged();
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/EnrichDocumentationIT.java]: this should be `response` not `deletePolicyResponse` methinks

**Review by hub-cap — APPROVED** [review]
I sure do love reviewing `AcknowledgedResponse` HLRC APIs :)

**Review by jbaiera — APPROVED** [review]
LGTM

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/EnrichDocumentationIT.java (hunk 1 above)
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/EnrichDocumentationIT.java]: good catch. I copied this from the put policy test and there the same mistake was made too :)

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**